### ISSUE TITLE
Help Wanted Sanzu Demo Deployment

### ISSUE THREAD

**Issue by rbah31** [cea-sec/sanzu#40]
Hello,

I have some issues with Sanzu deployment on my machine. I am trying to launch the demo on my Debian VM. After installing and building the project, I try to run the file run-demo.sh but I have an unbound variable error. I maybe missed some steps on the deployment.

<img width="1440" alt="Issue Terminal" src="https://user-images.githubusercontent.com/81626365/175251673-0bc49d76-eaa9-41be-b44d-4556b80ef937.png">
 
Could you help me to resolve this problem ? Thanks !

**Comment by serpilliere** [comment]
Hi @rbah31 
The demo runs the client and the server on docker, both running on the local machine.
The client uses the current DISPLAY to display the server video so you must have the DISPLAY variable set to the current DISPLAY used by your actual graphic session.
Maybe you don't have the DISPLAY variable set ?
For example, if you launch the demo from a ssh client, the DISPLAY may not be set.

**Comment by rbah31** [comment]
> Hi @rbah31 The demo runs the client and the server on docker, both running on the local machine. The client uses the current DISPLAY to display the server video so you must have the DISPLAY variable set to the current DISPLAY used by your actual graphic session. Maybe you don't have the DISPLAY variable set ? For example, if you launch the demo from a ssh client, the DISPLAY may not be set.

Hi, 
Indeed it was the problem, thank you ! I was wondering if you provide to add documentations about Sanzu deployment (some help concerning deployment step by step would be appreciated :) )

**Comment by rbah31** [comment]
Hi,

I still have some issues concerning the deployment... I am using a minimal VM that's possibly why I have so much issues. 
I have an OS error 111 when I try to deploy the demo.

<img width="1440" alt="OS Error 111" src="https://user-images.githubusercontent.com/81626365/175917633-71c5e4a1-97e9-483c-8a4c-8f1c9274ad63.png">

I don't know how I can fix it. Thank you in advance for your answer.

**Comment by serpilliere** [comment]
Ok it seems it's the same problem.
Let me rephrase the readme. If you have a VM, let say "carapuce", and have your machine, let say "bulbizar" on which you want to have the display of caprapuce. Here are the steps to do that:
- get the carapuce display. If you have a x server running, "echo $DISPLAY". If no, you have to create a x11 server on carapuce. For example, with "Xvfb :1234". Then run your x11 client on this display. For example `DISPLAY=:1234 xfwm4`
- now execute the sanzu_server on carapuce to export this display: `DISPLAY=:1234 sanzu_server -f sanzu.toml`
- now, on you local machine, bulbizar, execute the client using your *local* display (for example :1) (to get it, execute "echo $DISPLAY" on bulbizar)
- run the client with `DISPLAY=:1 sanzu_client bulbizar_ip 1122` (replace *bulbizar_ip* by bulbizar ip)

If you have no display on your local machine (carapuce) you won't be able to launch sanzu, as it's a graphical remote display.


If you have a special case, maybe you can describe us your configuration (local and remote) such that we may help you.

**Comment by rbah31** [comment]
Thank you for your help, I have now a better understanding of Sanzu and how it works. 

I don't know if I should open another issue for this question, but I was wondering how does your VM cluster work. Indeed, I'm trying to have a better understanding of the structure but I didn't find something in the source code concerning the call of VM or something similar yet. 

I saw in SSTIC acts that you are using a VMM and that there is a script run by the broker that took a free VM in a pool of unassigned VMs already started and once it's done, the VMM re-runs another VM so I was wondering what VMM did you use and how do you proceed to create this VM pool (VM templating...) ?

**Comment by serpilliere** [comment]
Hi @rbah31 
I am closing this issue and opening a new one on this specific subject, as it deserves its own thread :smile:
